lib/reek/examiner.rb

```diff
@@ -97,12 +97,15 @@ def smelly?
     #
     # :reek:TooManyStatements { max_statements: 8 }
     def run
-      return [] unless syntax_tree
-      begin
-        examine_tree
-      rescue Errors::BaseError => exception
-        raise unless @error_handler.handle exception
-        []
+      if source.valid_syntax? && syntax_tree
```

Comment:
Yeah, I also think this was for empty files. Apparently we don't have specs for this?